catalyst_voices_packages/catalyst_compression/catalyst_compression_web/lib/src/interop/catalyst_compression_interop.dart

```diff
@@ -65,20 +70,24 @@ class JSZstdCompressor implements CatalystCompressor {
   const JSZstdCompressor();
 
   @override
-  List<int> compress(List<int> bytes) {
+  Future<List<int>> compress(List<int> bytes) async {
     if (bytes.length < _minLength) {
       throw CompressionNotSupportedException(
         'Bytes too short, actual: ${bytes.length}, required: $_minLength',
       );
     }
 
-    final data = zstdCompress(hex.encode(bytes).toJS).toDart;
-    return hex.decode(data);
+    final data = await promiseToFuture<JSString>(
+      zstdCompress(hex.encode(bytes).toJS),
+    );
+    return hex.decode(data.toDart);
   }
 
   @override
-  List<int> decompress(List<int> bytes) {
-    final data = zstdDecompress(hex.encode(bytes).toJS).toDart;
-    return hex.decode(data);
+  Future<List<int>> decompress(List<int> bytes) async {
+    final data = await promiseToFuture<JSString>(
+      zstdDecompress(hex.encode(bytes).toJS),
+    );
```

Comment:
A standard way to do it is to use `toDart` method. Make sure to check other places in this file for this pattern

```suggestion
    final data = await zstdDecompress(hex.encode(bytes).toJS).toDart;
```